### ISSUE TITLE
eslint-plugin-lodash-fp no-chain rule false positive on filter

### ISSUE THREAD

**Issue by don-motorefi** [jfmengels/eslint-plugin-lodash-fp#85]
Expected behavior:

This rule should be triggered when chaining functional calls outside of a `flow` or `compose` block.

Actual behavior:

The rule is triggered in some simple cases where no chaining is being used at all.

Reproduction case:

```
import filter from 'lodash'

  const doFilter = (value: string) => {
    const leadsFiltered = filter(leads, (item) => {
      return item.status === value
    })
  }
```

This result in the eslint rule errors:
```
Unallowed use of chain operations. Use flow/compose insteadeslint(lodash-fp/no-chain)
Prefer composition to Lodash chaining .   eslintlodash/chaining
```
